pkg/serverless/trace/propagation/carriers.go

```diff
@@ -286,9 +289,53 @@ func extractTraceContextFromStepFunctionContext(event events.StepFunctionPayload
 	return tc, nil
 }
 
-// stringToDdSpanID hashes the Execution ARN, state name, and state entered time to generate a 64-bit span ID
-func stringToDdSpanID(execArn string, stateName string, stateEnteredTime string) uint64 {
-	uniqueSpanString := fmt.Sprintf("%s#%s#%s", execArn, stateName, stateEnteredTime)
+// extractTraceContextFromNestedStepFunctionContext extracts the root Step Function's execution ARN to generate the Trace ID
+func extractTraceContextFromNestedStepFunctionContext(event events.NestedStepFunctionPayload) (*TraceContext, error) {
+	if event.RootExecutionID == "" {
+		return nil, errorNoStepFunctionContextFound
+	}
+
+	tc, err := extractTraceContextFromStepFunctionContext(event.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	tc.TraceID, tc.TraceIDUpper64Hex = stringToDdTraceIDs(event.RootExecutionID)
+
+	return tc, nil
+}
+
+// extractTraceContextFromLambdaRootStepFunctionContext extracts the explicitly defined Trace ID and uses that value
+func extractTraceContextFromLambdaRootStepFunctionContext(event events.LambdaRootStepFunctionPayload) (*TraceContext, error) {
+	if event.TraceID == "" || event.TraceTags == "" {
+		return nil, errorNoStepFunctionContextFound
+	}
+
+	tc, err := extractTraceContextFromStepFunctionContext(event.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	traceID, err := strconv.ParseUint(event.TraceID, 10, 64)
```

Comment:
nit. You _could_ just change this to 
```suggestion
	tc.TraceID, err = strconv.ParseUint(event.TraceID, 10, 64)
```